### ISSUE TITLE
After Disaster Recovery CephFS PVC doesn't work

### ISSUE THREAD

**Issue by Krast76** [rook/rook#8442]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

After following this documentation : https://github.com/rook/rook/blob/master/Documentation/ceph-disaster-recovery.md#steps the ceph cluster was back and all RBD volumes and associated PVC work well. But all cephfs PVC doesn't work.

**Expected behavior:**

Working CephFS PVC after Disaster Recovery

**How to reproduce it (minimal and precise):**

Following these steps : https://github.com/rook/rook/blob/master/Documentation/ceph-disaster-recovery.md#steps on a cluster with CephFS PVC volumes in use.

**File(s) to submit**:

On a pod mounting a CephFS PVC i see this error log : 

```bash
MountVolume.MountDevice failed for volume "pvc-60a45634-ffde-4db2-b825-7947c0cf6440" : rpc error: code = Internal desc = fscID (1) not found in Ceph cluster
```

On the csi-cephfsplugin i see this error log (many times) : 

```
E0727 13:34:12.891419       1 utils.go:163] ID: 420 Req-ID: 0001-0009-rook-ceph-0000000000000001-379c24fb-ba17-11eb-b860-4a6050610914 GRPC error: rpc error: code = Internal desc = fscID (1) not found in Ceph cluster
```

In fact, ceph-csi is looking for the fscID (FileSystem ID) 1 but after recovering my cephfs got the id 2.

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.5 LTS (Bionic Beaver)
* Kernel (e.g. `uname -a`): `4.15.0-122-generic #124-Ubuntu SMP Thu Oct 15 13:03:05 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: on-premise
* Rook version (use `rook version` inside of a Rook Pod): rook/ceph:v1.3.9 (but the behavior should be the same on all versions I think)
* Storage backend version (e.g. for ceph do `ceph -v`): 14.2.10-0
* Kubernetes version (use `kubectl version`):  v1.19.9
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Deployed with kubeadm (via kubespray)
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): HEALTH_OK (after the recovering)

**Comment by travisn** [comment]
@Rakshith-R What might be missing in the disaster recovery guide for the correct fscID for the cephfs volumes?

**Comment by Madhu-1** [comment]
This is true for RBD also. the RBD pool needs to be recovered with the same pool ID.

**Comment by Rakshith-R** [comment]
> @Rakshith-R What might be missing in the disaster recovery guide for the correct fscID for the cephfs volumes?

I did find how to preserve/recover same cephfs fscid https://docs.ceph.com/en/latest/cephfs/recover-fs-after-mon-store-loss/ but not sure at which step it should be done. 
(@Krast76 , just in case if you were still interested)


> This is true for RBD also. the RBD pool needs to be recovered with the same pool ID.

Did'nt find any docs for this on ceph side (unlikely but maybe pool id is reserved ?).

**Comment by travisn** [comment]
> This is true for RBD also. the RBD pool needs to be recovered with the same pool ID.

@Madhu-1 The pool id is assigned by Ceph in the order the pools are created, which would make it very difficult to predictably re-create pools with the same ID during disaster recovery. [This](https://access.redhat.com/solutions/3398411) indicates the same, and neither do I see anything in the [pool docs](https://docs.ceph.com/en/latest/rados/operations/pools/) about pool ids being configurable. Any other possibilities?

**Comment by stevefan1999-personal** [comment]
Please reopen this issue. Now, I have a slightly different setup that my fscID is now 1 instead of what is expected by the CSI, 2. So I want to find a way to reset the fscID in the CSI.

**Comment by stevefan1999-personal** [comment]
I instead followed the steps [here](https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/). I modified toolbox to run this

**Comment by stevefan1999-personal** [comment]
I tried to create a mapping in rook-ceph-csi-mapping-config as key `csi-mapping-config-json`
```json
[{"ClusterIDMapping":{"rook-ceph":"rook-ceph"},"RBDPoolIDMapping":null,"CephFSFscIDMapping": [{ "2": "1" }]}]
```
However it will get reset ever ytime I restart the provisioner

**Comment by stevefan1999-personal** [comment]
so, this is one of my volume ID captured:
```
0001-0009-rook-ceph-0000000000000002-41e63e04-2c41-11ec-a5e9-2a0582d093b3
```
as you can see, because the location id is set to 2, this obviously coincide with the code here:
https://github.com/ceph/ceph-csi/blob/0bd1d44dc07bc9ec3762a3f966dd501b4e382262/internal/cephfs/core/volumeoptions.go#L275-L283

@travisn wonder if you wanted to take a look?

**Comment by travisn** [comment]
@Madhu-1 Is there any way to apply the remapping of the pool IDs like that and have it stick?

**Comment by Madhu-1** [comment]
it's only available for rbd now. it's not implemented for cephfs. @stevefan1999-personal one option is to recreate the PV with new volumeHandle which has the correct filesystem ID.
